File: examples/bimapi/CheckbotBimLink/FeaExample/BimApiLinkFeaExample/Importers/LoadCaseImporter.cs
using BimApiLinkFeaExample.FeaExampleApi;
using IdeaRS.OpenModel.Loading;
using IdeaStatiCa.BimApi;
using IdeaStatiCa.BimApiLink.BimApi;
using IdeaStatiCa.BimApiLink.Importers;

namespace BimApiLinkFeaExample.Importers
{
	internal class LoadCaseImporter : IntIdentifierImporter<IIdeaLoadCase>
	{
		private readonly IFeaLoadsApi loadsApi;

		public LoadCaseImporter(IFeaLoadsApi loadsApi)
		{
			this.loadsApi = loadsApi;
		}

		public override IIdeaLoadCase Create(int id)
		{
			IFeaLoadCase loadCase = loadsApi.GetLoadCase(id);
			var loadType = GetLoadType(loadCase.LoadCaseType);
			return new IdeaLoadCase(id)
			{
				Name = loadCase.Name,
				LoadGroup = Get<IIdeaLoadGroup>(loadCase.LoadGroupId),
				LoadType = loadType.Item1,				
				Type = loadType.Item2,
				Variable = VariableType.Standard
			};
		}

		private static (LoadCaseType, LoadCaseSubType) GetLoadType(TypeOfLoadCase typeOfLoadCase) 
		{
			switch (typeOfLoadCase)
			{
				case TypeOfLoadCase.Selfweight:
					return (LoadCaseType.Permanent, LoadCaseSubType.PermanentSelfweight);
				case TypeOfLoadCase.DeadLoad:
					return (LoadCaseType.Permanent, LoadCaseSubType.PermanentStandard);
				case TypeOfLoadCase.Snow:
					return (LoadCaseType.Variable, LoadCaseSubType.VariableStatic);
				default:
					return (LoadCaseType.Permanent, LoadCaseSubType.PermanentStandard);
			}
		}
	}
}

File: examples/bimapi/CheckbotBimLink/CppFeaExample/CppFeaApiWrapper/LoadCaseImporter.cpp
#include "pch.h"
#include "LoadCaseImporter.h"
#include "..\CppFeaApi\NativeFeaLoads.h"

namespace CppFeaApiWrapper
{
	namespace Importers
	{
		LoadCaseImporter::LoadCaseImporter(ImporterContext^ context)
		{
			this->context = context;
		}

		IIdeaLoadCase^ LoadCaseImporter::Create(int id)
		{
			auto feaLoadCase = this->context->GetApi()->GetLoading()->GetLoadCase(id);

			auto loadType = GetLoadType(feaLoadCase->LoadCaseType);

			IdeaLoadCase^ ideaLoadCase = gcnew IdeaLoadCase(id);

			ideaLoadCase->Name = gcnew String(feaLoadCase->Name.c_str());
			ideaLoadCase->LoadGroup = Get<IIdeaLoadGroup^>(feaLoadCase->LoadGroupId);
			ideaLoadCase->LoadType = loadType->Item1;
			ideaLoadCase->Type = loadType->Item2;
			ideaLoadCase->Variable = VariableType::Standard;

			return ideaLoadCase;
		}

		Tuple<LoadCaseType, LoadCaseSubType>^ LoadCaseImporter::GetLoadType(int typeOfLoadCase)
		{
			switch (typeOfLoadCase) {
			case NativeFeaLoads::LoadCase_Type_Selfweight:
				return gcnew Tuple<LoadCaseType, LoadCaseSubType>(IdeaRS::OpenModel::Loading::LoadCaseType::Permanent, IdeaRS::OpenModel::Loading::LoadCaseSubType::PermanentSelfweight);
			case NativeFeaLoads::LoadCase_Type_Dead:
				return gcnew Tuple<LoadCaseType, LoadCaseSubType>(IdeaRS::OpenModel::Loading::LoadCaseType::Permanent, IdeaRS::OpenModel::Loading::LoadCaseSubType::PermanentStandard);
			case NativeFeaLoads::LoadCase_Type_Snow:
				return gcnew Tuple<LoadCaseType, LoadCaseSubType>(IdeaRS::OpenModel::Loading::LoadCaseType::Variable, IdeaRS::OpenModel::Loading::LoadCaseSubType::VariableStatic);
			default:
				return gcnew Tuple<LoadCaseType, LoadCaseSubType>(IdeaRS::OpenModel::Loading::LoadCaseType::Permanent, IdeaRS::OpenModel::Loading::LoadCaseSubType::PermanentStandard);
			}
		}
	}
}
